pkg/controllers/hpareplicassyncer/hpa_replicas_syncer_predicate.go

```diff
@@ -0,0 +1,79 @@
+package hpareplicassyncer
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+)
+
+const KindHPA = "HorizontalPodAutoscaler"
```

Comment:
We could use this to replace:  

https://github.com/karmada-io/karmada/blob/8d1a8007ebdbd97a2fe12cf8ee7d4694cfb2ed82/pkg/util/constants.go#L108